src/test/java/uk/gov/laa/ccms/caab/bean/validators/client/ClientAddressDetailsFindAddressValidatorTest.java

```diff
@@ -0,0 +1,128 @@
+  public void setUp() {
+    clientDetails = buildClientDetails();
+    errors = new BeanPropertyBindingResult(clientDetails, "clientDetails");
+  }
+
+  @Test
+  public void supports_ReturnsTrueForClientDetailsClass() {
+    assertTrue(clientAddressDetailsFindAddressValidator.supports(ClientDetails.class));
+  }
+
+  @Test
+  public void supports_ReturnsFalseForOtherClasses() {
+    assertFalse(clientAddressDetailsFindAddressValidator.supports(Object.class));
+  }
+
+  @Test
+  public void validate() {
+    clientAddressDetailsFindAddressValidator.validate(clientDetails, errors);
+    System.out.println(errors.getAllErrors());
+    assertFalse(errors.hasErrors());
+  }
+
+  @ParameterizedTest
+  @NullAndEmptySource
+  public void validate_countryRequired(String country) {
+    clientDetails.setCountry(country);
+    clientAddressDetailsFindAddressValidator.validate(clientDetails, errors);
+    assertTrue(errors.hasErrors());
+    assertNotNull(errors.getFieldError("country"));
+    assertEquals("required.country", errors.getFieldError("country").getCode());
+  }
+
+  @Test
+  public void validate_countryNotUnitedKingdom() {;
+    clientDetails.setCountry("USA");
+    clientAddressDetailsFindAddressValidator.validate(clientDetails, errors);
+    assertTrue(errors.hasErrors());
+    assertNotNull(errors.getFieldError("country"));
+    assertEquals("required.GBR", errors.getFieldError("country").getCode());
+  }
+
+  @ParameterizedTest
+  @NullAndEmptySource
+  public void validate_houseNameNumberRequired(String houseNameNumber) {
+    clientDetails.setHouseNameNumber(houseNameNumber);
+    clientAddressDetailsFindAddressValidator.validate(clientDetails, errors);
+    assertTrue(errors.hasErrors());
+    assertNotNull(errors.getFieldError("houseNameNumber"));
+    assertEquals("required.houseNameNumber", errors.getFieldError("houseNameNumber").getCode());
+  }
+
+  @ParameterizedTest
+  @CsvSource({
+      "country, test",
+      "houseNameNumber, test",
+      "postcode, SW1A 1AA",
+      "addressLine1, test",
+      "addressLine2, test",
+      "cityTown, test",
+      "county, test"
+  })
+  public void validate_noFixedAbode_invalid(String field, String value) {
+    clientDetails = new ClientDetails();
+    clientDetails.setNoFixedAbode(true);
+
+    if (field.equals("country")) {
+      clientDetails.setCountry(value);
+    } else if (field.equals("houseNameNumber")) {
+      clientDetails.setHouseNameNumber(value);
+    } else if (field.equals("postcode")) {
+      clientDetails.setPostcode(value);
+    } else if (field.equals("addressLine1")) {
+      clientDetails.setAddressLine1(value);
+    } else if (field.equals("addressLine2")) {
+      clientDetails.setAddressLine2(value);
+    } else if (field.equals("cityTown")) {
+      clientDetails.setCityTown(value);
+    } else if (field.equals("county")) {
+      clientDetails.setCounty(value);
+    }
+
+    clientAddressDetailsFindAddressValidator.validate(clientDetails, errors);
```

Comment:
should we also be asserting how many errors we're expecting, or that too much detail for these tests?

Reply:
makes sense to include that

Reply:
amended